src/analyzer/analyzersilence.cpp

```diff
@@ -0,0 +1,106 @@
+#include "analyzer/analyzersilence.h"
+#include "util/audiosignal.h"
+
+namespace {
+    const SINT kChannelCount = mixxx::AudioSignal::kChannelCountStereo;
+    const float kSilenceThresholdDb = -60.0f;
+}  // anonymous namespace
+
+AnalyzerSilence::AnalyzerSilence(UserSettingsPointer pConfig)
+    : m_pConfig(pConfig),
+      m_fThreshold(db2ratio(kSilenceThresholdDb)),
+      m_iFramesProcessed(0),
+      m_bPrevSilence(true),
+      m_iSignalBegin(-1),
+      m_iSignalEnd(-1) {
+}
+
+AnalyzerSilence::~AnalyzerSilence() {
+}
+
+bool AnalyzerSilence::initialize(TrackPointer tio, int sampleRate, int totalSamples) {
+    Q_UNUSED(sampleRate);
+
+    m_iFramesProcessed = 0;
+    m_bPrevSilence = true;
+    m_iSignalBegin = -1;
+    m_iSignalEnd = -1;
+
+    m_pBeginCue = tio->findCueByType(Cue::BEGIN);
+    if (!m_pBeginCue) {
+        m_pBeginCue = tio->createAndAddCue();
+        m_pBeginCue->setType(Cue::BEGIN);
+        m_pBeginCue->setSource(Cue::AUTOMATIC);
+        m_pBeginCue->setPosition(0);
+    }
+
+    m_pEndCue = tio->findCueByType(Cue::END);
+    if (!m_pEndCue) {
+        m_pEndCue = tio->createAndAddCue();
+        m_pEndCue->setType(Cue::END);
+        m_pEndCue->setSource(Cue::AUTOMATIC);
+        m_pEndCue->setPosition(totalSamples);
+    }
+
+    return !isDisabledOrLoadStoredSuccess(tio);
+}
+
+bool AnalyzerSilence::isDisabledOrLoadStoredSuccess(TrackPointer tio) const {
+    Q_UNUSED(tio);
+
+    return m_pBeginCue->getSource() == Cue::MANUAL && m_pEndCue->getSource() == Cue::MANUAL;
+}
+
+void AnalyzerSilence::process(const CSAMPLE *pIn, const int iLen) {
```

Comment:
Regarding [coding guidelines](https://mixxx.org/wiki/doku.php/coding_guidelines):
    The “*” should be aligned with the type and not the name. 
please change `const CSAMPLE *pIn` to `const CSAMPLE* pIn`.